fili-core/src/main/java/com/yahoo/bard/webservice/logging/RequestLog.java

```diff
@@ -35,79 +22,76 @@
+
 /**
- * Represents the logging framework that provides timing capabilities of arbitrary phases on the handling lifecycle of a
- * request and accumulation of information for such a request in a single mega log line.
+ * The RequestLog holds data that we would like to log about the request. In particular, various timings are
+ * accumulated here.
  */
 public class RequestLog {
-
+    private static final MetricRegistry REGISTRY = MetricRegistryFactory.getRegistry();
     private static final Logger LOG = LoggerFactory.getLogger(RequestLog.class);
-    private static final SystemConfig SYSTEM_CONFIG = SystemConfigProvider.getInstance();
-
-    private static final String ID_KEY = "logid";
     private static final long MS_PER_NS = 1000000;
-    private static final MetricRegistry REGISTRY = MetricRegistryFactory.getRegistry();
-    private static final ThreadLocal<RequestLog> RLOG = ThreadLocal.withInitial(RequestLog::new);
-    private static final String LOGINFO_ORDER_STRING = SYSTEM_CONFIG.getStringProperty(
-            SYSTEM_CONFIG.getPackageVariableName("requestlog_loginfo_order"), ""
-    );
-    private static final List<String> LOGINFO_ORDER = generateLogInfoOrder(LOGINFO_ORDER_STRING);
+    protected static final String LOG_ID_KEY = "logid";
 
-    private String logId;
-    private LogBlock info;
-    private TimedPhase mostRecentTimer;
-    private final Map<String, TimedPhase> times;
-    private final Set<String> threadIds;
+    protected String logId;
+    protected LogBlock info;
+    protected TimedPhase mostRecentTimer;
+    protected final Map<String, TimedPhase> times;
+    protected final Set<String> threadIds;
 
     /**
      * This class has only static methods and is not supposed to be directly instantiated.
      */
-    private RequestLog() {
+    protected RequestLog() {
         logId = null;
         info = null;
         mostRecentTimer = null;
         times = new LinkedHashMap<>();
         threadIds = new LinkedHashSet<>();
-        MDC.remove(ID_KEY);
+        MDC.remove(LOG_ID_KEY);
+        init();
     }
 
     /**
      * Copy constructor is also private.
      *
      * @param  rl request log object to copy from
      */
-    private RequestLog(RequestLog rl) {
+    protected RequestLog(RequestLog rl) {
         logId = rl.logId;
         info = rl.info;
         mostRecentTimer = rl.mostRecentTimer;
         times = new LinkedHashMap<>(rl.times);
         threadIds = new LinkedHashSet<>(rl.threadIds);
-        MDC.put(ID_KEY, logId);
+        MDC.put(LOG_ID_KEY, logId);
     }
 
     /**
      * Represents a phase that is timed.
      * TimedPhase is used to associate a Timer located in the registry with the exact duration of such a phase for a
      * specific request.
      */
-    private static class TimedPhase {
-        private final String name;
-        private long start;
-        private long duration;
+    protected static class TimedPhase {
```

Comment:
Well, erm, hmm. I actually have a PR out to address a separate issue that pulls the `TimedPhase` out: https://github.com/yahoo/fili/pull/143

Basically, to make starting and stopping timers cleaner, I had the `TimedPhase` implement auto-closeable, so that we could use Java's spiffy new try-with-resource block. But that required making the `TimedPhase` publicly available, and I'm not generally a fan of having public inner classes vs. just regular public classes. Don't really believe that they buy you much but code clutter.